lighthouse-treemap/app/src/main.js

```diff
@@ -78,6 +78,8 @@ class TreemapViewer {
     this.viewModes;
     /** @type {RenderState=} */
     this.previousRenderState;
+    /** @type {WeakMap<HTMLElement, LH.Treemap.Node>} */
```

Comment:
Out of curiosity, why a `WeakMap` if it's recreated with every new version of the table? To me it still signals something special is going on so be careful while reading, but it's probably actually a good default to use when you don't need to iterate and perf isn't a big concern...

Reply:
In addition to different bundle selector values re-creating the table, scrolling also recreates rows. They are all lazily rendered. So it's good that scrolling doesn't eventually result in OOM :)

Reply:
also agree it's just a good default